subsys/bluetooth/audio/ascs.c

```diff
@@ -2810,6 +2828,76 @@ static ssize_t ascs_cp_write(struct bt_conn *conn,
 respond:
 	control_point_notify(conn, rsp_buf.data, rsp_buf.len);
 
+	ascs_event_cp_write_destroy(evt);
+}
+
+static void ascs_event_handler(struct ascs_event *event)
+{
+	switch (event->type) {
+	case ASCS_EVENT_CP_WRITE:
+		return handle_event_cp_write(&event->cp_write);
+	}
+}
+
+static void ascs_thread(void)
+{
+	struct ascs_event event;
+
+	while (true) {
+		k_msgq_get(&ascs_msgq, &event, K_FOREVER);
+		ascs_event_handler(&event);
+	}
+}
+
+NET_BUF_POOL_DEFINE(cp_write_buf_pool, ASCS_MAX_CP_WRITES, BT_ATT_BUF_SIZE, 0, NULL);
```

Comment:
As the Control Point write operation can be GATT Long Write procedure, the size of this buffer is not sufficient to handle it.